File: An 2/Sem 1/MAP/Final Templates/Csharp Exam/Csharp Exam/Repository/IRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Csharp_Exam.Repository
{
    interface IRepository<T>
    {
        void addElement(T element);
        void removeElement(T element);
        List<T> allElements();
    }
}

File: An 2/Sem 1/MAP/Final Templates/Java Exam/src/com/company/Repository/IRepository.java
package com.company.Repository;

import java.util.List;

/**
 * Created by C.Mihai on 01/02/14.
 */
public interface IRepository<T> {
    public void addElement(T element);
    public void removeElement(T element);
    public List<T> allElements();
}
